pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -1403,3 +1405,79 @@ func TestReadPreNUMACheckpoint(t *testing.T) {
 	err = m.readCheckpoint()
 	require.NoError(t, err)
 }
+
+func TestGetTopologyHintsWithUpdates(t *testing.T) {
+	socketDir, socketName, _, err := tmpSocketDir()
+	defer os.RemoveAll(socketDir)
+	require.NoError(t, err)
+
+	devs := []pluginapi.Device{}
+	for i := 0; i < 1000; i++ {
+		devs = append(devs, pluginapi.Device{
+			ID:     fmt.Sprintf("dev-%d", i),
+			Health: pluginapi.Healthy,
+			Topology: &pluginapi.TopologyInfo{
+				Nodes: []*pluginapi.NUMANode{
+					{ID: 0},
+				},
+			}})
+	}
+	topology := []cadvisorapi.Node{
+		{Id: 0},
+	}
+	m, _ := setupDeviceManager(t, nil, nil, socketName, topology)
+	mimpl := m.(*wrappedManagerImpl)
+
+	testCases := []struct {
+		description string
+		count       int
+		devices     []pluginapi.Device
+		testfunc    func() (r interface{})
+	}{
+		{
+			description: "getAvailableDevices data race when update device",
+			count:       1,
+			devices:     devs,
+			testfunc: func() (r interface{}) {
+				mimpl.getAvailableDevices(testResourceName)
+				return nil
+			},
+		},
+		{
+			description: "generateDeviceTopologyHints data race when update device",
+			count:       1,
+			devices:     devs,
+			testfunc: func() (r interface{}) {
+				mimpl.generateDeviceTopologyHints(
+					testResourceName, sets.NewString(), sets.NewString(), 1)
+				return nil
+			},
+		},
+		{
+			description: "deviceHasTopologyAlignment data race when update device",
+			count:       1000,
+			devices:     devs[:1],
+			testfunc: func() (r interface{}) {
+				mimpl.deviceHasTopologyAlignment(testResourceName)
+				return nil
+			},
+		},
+	}
+
+	updated := atomic.Bool{}
+	for _, test := range testCases {
+		updated.Store(false)
```

Comment:
run inside t.Run() to use subtests and avoid test pollution

Reply:
it may require that you move the initialization inside the t.Run?

Reply:
Thanks. I modified it with subtests, and the code for initializing the manager was moved into the subtest.